### DISCUSSION TITLE
Not reading my .env file

### DISCUSSION THREAD

**Discussion by tausyy** [caarlos0/env#182] [Q&A] [answered]
I have my .env file with 2 properties

```
BOT_TOKEN=rightHereIsWhereMyTokenIs
TEST=1234
```

when I `go run main.go` with this code this is my output

```
env: environment variable "BOT_TOKEN" should not be empty
{BotToken: Test:0}
```

What am I doing wrong?

```golang
package main

import (
	"fmt"

	"github.com/caarlos0/env/v6"
)

type config struct {
	BotToken string `env:"BOT_TOKEN,notEmpty"`
	Test     int32  `env:"TEST"`
}

func main() {
	cfg := config{}
	if err := env.Parse(&cfg); err != nil {
		fmt.Printf("%+v\n", err)
	}

	fmt.Printf("%+v\n", cfg)
}
```

**Comment by caarlos0** [comment, marked as answer]
This lib does not handle `.env` files. It only parses environment variables to an user-defined struct.

**Reply by pedromassango** [reply]
I think this sould be made clear in the docs of this lib.

**Reply by caarlos0** [reply]
it doesn't say anything about env files at all ¯\_(ツ)_/¯

**Reply by pedromassango** [reply]
... and that is the problem 😄 

For someone who is just starting with GoLang and has no clue about this package he will definitely try it just to find out and it does not work. If we had that info in the readme file it would be a different story.

BTW, would it be possible to make it read from .env file and return a, let's say, EnvConfig struct?

**Reply by caarlos0** [reply]
> For someone who is just starting with GoLang and has no clue about this package he will definitely try it just to find out and it does not work. If we had that info in the readme file it would be a different story.

I understand, but if we start adding everything it doesn't do but users could think it does in the readme, it would get quite big...

> BTW, would it be possible to make it read from .env file and return a, let's say, EnvConfig struct?

there are another libs for this. This one only parses environment variables into structs.

**Comment by svalls26** [comment]
Hey! Just found this discussion and the same happened to me, when reading the readme gives the impression that it is going to take the vars directly from your .env file. I'm new in Go, and I figured it out because other coworker told me I need to load the .env with other pckg. 
I understand the worries about having a big readme but I definitely think it's worth to mentioned that the .env file needs to be loaded apart in order to use it.

**Reply by caarlos0** [reply]
which part of the readme gave you that impression?

there's no mention of env files anywhere...

**Reply by andradei** [reply]
You're right about the README.
I think people are getting the "wrong impression" because that's such a common feature it's pretty much assumed.

This package would really be complete if it had the option to read from and `.env` file.

**Comment by IlmirSharifullin** [comment]
I've also ran into this problem xD. You, guys, can use github.com/joho/godotenv (or its autoload) module together with this one.
Something like:
`import (
	"fmt"
	"github.com/caarlos0/env/v10"
	_ "github.com/joho/godotenv/autoload"
)

type Config struct {
	BotToken string `env:"BOT_TOKEN"`
	// and more more
}

func main() {
	cfg := Config{}
	if err := env.Parse(&cfg); err != nil {
		fmt.Printf("%+v\n", err)
	}
	fmt.println(cfg.BotToken)
}`

**Comment by icarosuper** [comment]
In my case I managed to remove this weird character (even though I couldn't see it) by selecting this option, I'm using the GoLand IDE.

<img width="502" height="386" alt="Untitled" src="https://github.com/user-attachments/assets/84304c46-0f9d-4ebc-a8a9-2ac858f73675" />
